render/wgpu/shaders/gradient.frag

```diff
@@ -6,18 +6,27 @@ layout(set = 0, binding = 2) uniform Colors {
 };
 
 layout(std430, set = 0, binding = 3) buffer Gradient {
+    vec4 u_colors[16];
+    float u_ratios[16];
     int u_gradient_type;
     uint u_num_colors;
     int u_repeat_mode;
+    int u_interpolation;
```

Comment:
I feel like the smart thing to do would be to merge `u_gradient_type` and `u_repeat_mode` and `u_interpolation` into a bitmask and save us a few bytes & alignment issues. I don't know if it's worth doing right now though, this works!

Later, `u_colors` and `u_ratios` will either move into a texture or into a variable length `Record u_records[]`, which will need to be at the end of the array. But later :D

Reply:
Wouldn't the code to deal with a bitmask add more bytes than using a bitmask would save? I don't know too much about rust, so I could be completely wrong.

Reply:
I mean bytes in the buffer, which is more about memory transfer cost. In my experience, keep frequently transferred buffers tiny - but that's video games and I don't know how much that's a real world issue here yet.

It does mean more code, so bytes on filesize sure. That's not too much of an issue here. But for bytes in memory:
Let's say there's 100 gradients in a movie (a reasonable number I think?).
It costs 4 bytes per int, so it's 24 bytes per gradient to have individual type, repeat mode, and interpolation. For 100 gradients that's 2,400 bytes of memory. With only 1 int to contain all three, that's 400 bytes of memory.

It's trivial stuff and maybe it's too early to optimise such things, but it's an easy win to keep in mind!

Reply:
Oh, OK, that makes sense. I didn't realize that the buffer the flags were on would be duplicated so many times.